maps/src/main/java/org/odk/collect/maps/layers/OfflineMapLayersPicker.kt

```diff
@@ -116,6 +116,13 @@ class OfflineMapLayersPicker(
             dismiss()
         }
 
+        if (sharedViewModel.layersToImport.value?.value == null) {
```

Comment:
This is probably the right way to solve this, but it does feel like we're working against ourselves here a little to have to write specific code for this case.

Considering it more, I'm realizing an alternative would be to fully support this case with Fragment args - pass the list of URIs as an arg to `OfflineMapLayersImporter` (and then have it call `loadExistingLayers`) instead of communicating them via the ViewModel. I don't like this nearly as much, but it does feel like where Android is pushing us given we have state we want to this Fragment to be able between full recreations.

Another alternative (which wouldn't support full recreation) would be to rework the UI a little to not show `OfflineMapLayersImporter` until there `layersToImport` is available. This would prevent us from having to deal with the lifecycle problem at all. 

What do you think? Perhaps we could look at one of those as part of #6198 and go with this approach (just avoiding the crash) for now?

Reply:
> Another alternative (which wouldn't support full recreation) would be to rework the UI a little to not show OfflineMapLayersImporter until there layersToImport is available. This would prevent us from having to deal with the lifecycle problem at all.

This would require dismissing the confirmation dialog programmatically either way (that's the thing we don't like the most I guess). Plus I think it's better to display that dialog immediately and have the loading indicator there not in the first one (at least it makes more sense with the current UI).

I will take a look at the first option which was my first implementation and I liked it. If it doesn't require a lot of changes I will do that otherwise we can think about it later.

Reply:
Ok I still like option 1 but I think we should leave it as is for now (merge this pr) and see if it's still the best solution once we figure out how we want to rework the UI in https://github.com/getodk/collect/issues/6198